samcli/lib/build/workflow_config.py

```diff
@@ -305,13 +310,13 @@ class BasicWorkflowSelector:
     Basic workflow selector that returns the first available configuration in the given list of configurations
     """
 
-    def __init__(self, configs):
-        if not isinstance(configs, list):
-            configs = [configs]
-
-        self.configs = configs
+    def __init__(self, configs: Union[CONFIG, List[CONFIG]]) -> None:
+        if isinstance(configs, list):
+            self.configs = configs
+        else:
+            self.configs = [configs]
```

Comment:
[Minor] `self.configs = configs if isinstance(configs, list) else [config]` 

Also, is it `list` of `List` ?

Reply:
`list` and `List` both work, to keep it consistent, let's use `List` 👍 

```py
isinstance([], typing.List)
True
isinstance([], list)
True
```

Reply:
FYI: `List = _alias(list, T, inst=False)`

Reply:
I had to revert this 😢 pylint cannot recognize this alias

> W1116: Second argument of isinstance is not a type (isinstance-second-argument-not-valid-type)

Reply:
Why the logic change?